pkg/controller/eventing/inmemory/clusterprovisioner/reconcile.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterprovisioner
+
+import (
+	"context"
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/pkg/controller"
+	"github.com/knative/eventing/pkg/system"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	Name = "in-memory-channel"
+)
+
+type reconciler struct {
+	client   client.Client
+	recorder record.EventRecorder
+	logger   *zap.Logger
+}
+
+// Verify the struct implements reconcile.Reconciler
+var _ reconcile.Reconciler = &reconciler{}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	//TODO use this to store the logger and set a deadline
+	ctx := context.TODO()
+	logger := r.logger.With(zap.Any("request", request))
+
+	cp := &eventingv1alpha1.ClusterProvisioner{}
+	err := r.client.Get(ctx, request.NamespacedName, cp)
+
+	// The ClusterProvisioner may have been deleted since it was added to the workqueue. If so,
+	// there is nothing to be done.
+	if errors.IsNotFound(err) {
+		logger.Info("Could not find ClusterProvisioner", zap.Error(err))
+		return reconcile.Result{}, nil
+	}
+
+	// Any other error should be retried in another reconciliation.
+	if err != nil {
+		logger.Error("Unable to Get ClusterProvisioner", zap.Error(err))
+		return reconcile.Result{}, err
+	}
+
+	// Does this Controller control this ClusterProvisioner?
+	if !shouldReconcile(cp.Namespace, cp.Name) {
```

Comment:
We should also check if `cp.spec.Reconciles.Kind` is a `Channel`

Reply:
Done.